### DISCUSSION TITLE
Get well constraints

### DISCUSSION THREAD

**Discussion by Theophile88** [equinor/resdata#870] [Q&A] [answered]
Hi,
I would like to know how to access the well constraints (WCONPROD, WCONINJE) for wells at each time step.
Can it be done with ecl?
Thanks!

**Comment by lars-petter-hauge** [comment, marked as answer]
Hi,
Sorry for the delayed answer. No, I don't believe the simulators store the constraints of the wells in the summary files. One would have to get those from the `DATA` file instead. `ecl` does not parse the `DATA` file (to such an extent), perhaps though `opm` can help you out instead, that definitively parse the `DATA` file, I would try it!

Edit: I gave it a go to figure it out, but admittedly it was a bit difficult to find docs on it. `ecl2df` is slightly more user friendly in that regards (it uses `opm` to parse the data file, and provides a bit more convenience)

```
from ecl2df import wcon, EclFiles

eclfiles = EclFiles("MYDATADECK.DATA")
dframe = wcon.df(eclfiles)
```

And the Pandas dataframe you receive contains well constraint information

https://equinor.github.io/ecl2df/usage/wcon.html?highlight=wconprod

**Reply by Theophile88** [reply]
Hi,
 Thanks a lot for your reply! It is exactly what I was looking for. I did a script myself, but this option looks more complete and reliable.
